lib/testLauncher/composeConfig.js

```diff
@@ -45,26 +47,64 @@ const LOOKUP_DIRS = [
 ];
 
 /**
- * @description Recursive search upward in directory
- * @param {String} directory directory, from which search will be started
- * @param {Array} configFiles array of filenames to search
+ * @description Reads file from path and returns file as an object. Supports json, yaml, json5 and ini formats.
+ * @param {String} filePath path to file
  */
-function directorySearch(directory, configFiles) {
-	const files = fs.readdirSync(directory).reduce((res, file) => {
-		if (configFiles.includes(file)) {
-			const absolute = path.join(directory, file);
+function readConfigFile(filePath) {
+	const fileContent = fs.readFileSync(filePath, 'utf8');
+
+	switch (path.extname(filePath)) {
+		case '.yaml':
+		case '.yml':
+			return yaml.load(fileContent);
+		case '.json5':
+			return JSON5.parse(fileContent);
+		case '.ini':
+			return ini.parse(fileContent);
+		default:
+			try {
+				return JSON.parse(fileContent);
+			} catch (error) {
+				return ini.parse(fileContent);
+			}
+	}
+}
 
-			res.push(absolute);
-		}
+/**
+ * @description Recursive search and merging for Extends configs
+ * @param {object} defaultConfigObject main configuration file object
+ * @param {String} extendPath extend path of other configuration file to merge with
+ * @param {String} filePath currently in read configuration file path
+ * @param {Array} foundPaths array of extends paths. Used to verify circular dependency
+ */
+function findExtendConfigs(defaultConfigObject, extendPath, filePath, foundPaths) {
+	const pathToConfig = path.join(path.dirname(filePath), extendPath);
+
+	if (foundPaths.includes(pathToConfig)) {
+		throw new SuitestError(circularDependencyError(pathToConfig), 404);
```

Comment:
Seems that this code is rendundat here. Could you remove it pls?

Reply:
This code is not redundant. This part of code is responsible to check if extends have a circular recursion issue. If it is happens - it throws an error and stop further processing.